starlette/websockets.py

```diff
@@ -219,10 +200,9 @@ async def close(
         )
 
     async def send_response(self, response: Response) -> None:
-        if self._have_response_extension():
-            await response(self.scope, self.receive, self.send)
-        else:
-            await self.close(code=1008, reason=f"HTTP Response {response.status_code}")
+        if "websocket.http.response" in self.scope.get("extensions", {}):
+            return await response(self.scope, self.receive, self.send)
+        raise RuntimeError("The server doesn't support the WebSocket Denial extension.")
```

Comment:
For the record, I disagree here.  You are forcing the creation of boilerplate in the application.  When the server does not support the extension, a close() is the only available option.   This _requires_ the application to either always check for the presence of the extension, _or_ have a try_catch mechanism built in.  IMHO, sensible middleware should simply do the right thing, possibly issuing a warning.  Middleware like Starlette is supposed to  make the interaction with ASGI simpler, and remove such considerations from the application developer.

Reply:
I agree, this should fallback to `close()` if the extension isn't supported